tasks/main.yml

```diff
@@ -1,25 +1,134 @@
 # SPDX-License-Identifier: MIT
 ---
-- name: Set platform/version specific variables
-  include_tasks: tasks/set_vars.yml
-
-# Examples of some tasks:
 - name: Ensure required packages are installed
   package:
-    name: "{{ __template_packages }}"
+    name: "{{ __systemd_packages }}"
     state: present
 
-- name: Ensure required services are enabled and started
-  service:
+- name: Deploy unit files
+  copy:
+    src: "{{ item }}"
+    dest: "{{ __admin_units_dir }}/{{ item }}"
+    owner: root
+    group: root
+    mode: '0644'
+  when: systemd_unit_files is defined
+  with_items:
+    - "{{ systemd_unit_files }}"
+  register: __systemd_unit_files_result
+
+- name: Deploy unit file templates
+  template:
+    src: "{{ item }}.j2"
+    dest: "{{ __admin_units_dir }}/{{ item }}"
+    owner: root
+    group: root
+    mode: '0644'
+  when: systemd_unit_file_templates is defined
+  with_items:
+    - "{{ systemd_unit_file_templates }}"
```

Comment:
if `systemd_unit_file_templates` files have the `.j2` suffix you'll need to strip them off to use in `dest`

Reply:
never mind